### PR TITLE
Test YAML with PropTypes

### PR DESCRIPTION
**Pull request by meiamsome**
Moves to PropTypes in order to check for keys that shouldn't exist but do. Sadly, this also implements a lot of the wonderful work of @MeaningOf42 and I am sorry about that.

Also, uses describe to label the tests as parts of testing a file for cleaner output.

### CODE REVIEW

File: unit_testing/formats.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+})
+
+const video = module.exports.video = exact({
+  title: PropTypes.string.isRequired,
+  video_number: PropTypes.number.isRequired,
+  date: PropTypes.instanceOf(Date).isRequired,
+  video_id: PropTypes.string.isRequired,
+  repository: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.oneOf([false]),
```
Comment (meiamsome) [inline unit_testing/formats.js]: Can `repository` be false? I thought it shouldn't be, but we are missing a repository for Coding Challenge 19 and its file has a false here.

**Comment by MeaningOf42** [comment]
Wow was going to work on clearing up the output, but was putting it off because I didn't know how to do it. Glad you beat me too it :)
    It's great to see my work being improved upon. I'm currently traveling and don't have much time, but when I get back I'll be sure to read through the whole thing and see what I can learn.

File: _Courses/programming-with-text/11-chrome-extensions/11.1-introduction.md [hunk 2]
```diff
@@ -3,6 +3,7 @@ title: "Introduction to Chrome Extensions"
 video_number: 1
 date: 2017-11-13
 video_id: hkOTAmmuv_4
+repository: false,
```
Comment (nilsweb) [inline _Courses/programming-with-text/11-chrome-extensions/11.1-introduction.md]: There's an invalid comma.

File: _Courses/programming-with-text/11-chrome-extensions/11.5-pop-ups.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ video_number: 5
 date: 2017-11-21
 video_id: YQnRSa8MGwM
 can_contribute: true
+repository: false,
```
Comment (nilsweb) [inline _Courses/programming-with-text/11-chrome-extensions/11.5-pop-ups.md]: There's an invalid comma.

**Pull request merged**